docs/internal/continuous-profiler.md

```diff
@@ -0,0 +1,255 @@
+
+### Requirements
+
+* .NET 6.0 or higher (`ICorProfilerInfo12` available in runtime). Technically
+  it could be `ICorProfiler10`, and .NET Core 3.1/.NET 5.0, but these versions
+  are not supported by OpenTelemetry .NET AutoInstrumentation.
+* .NET Framework is not supported. `ICorProfiler10` nor `ICorProfiler12`
+  are not available in .NET Fx.
+
+### Enable the profiler
+
+Implement custom plugin.
+
+### Configuration settings by plugin
+
+* `threadSamplingEnabled = true;`
+* `var threadSamplingInterval = 10000u;` // in ms. Splunk is using 10000.
+  There is no possibility to set value lower than 1000.
+* `var exportInterval = TimeSpan.FromMilliseconds(500);` // Interval to read data
+  from buffers and call exporter, common for Thread and Allocation sampling
+* `object continuousProfilerExporter = new ConsoleExporter();` // Exporter,
+  common for Thread and Allocation sampling
+
+### Escape hatch
+
+The profiler limits its own behavior when both buffers used to store sampled
+data are full.
+
+This scenario might happen when the data processing thread is not able
+to export data the given period of time.
+
+Thread sampler will resume when any of the buffers are empty.
+
+### Troubleshooting the .NET profiler
+
+#### How do I know if it's working?
+
+At the startup, the SignalFx Instrumentation for .NET logs the string
+`ContinuousProfiler::StartThreadSampling` at `info` log level.
+
+You can grep for this in the native logs for the instrumentation
+to see something like this:
+
+```text
+10/12/22 12:10:31.962 PM [12096|22036] [info] ContinuousProfiler::StartThreadSampling
+```
+
+#### How can I see Continuous Profiling configuration?
+
+The OpenTelemetry .NET AutomaticInstrumentation logs the profiling configuration
+at `Debug` log level during the startup. You can grep for the string
+`Continuous profiling configuration:` to see the configuration.
+
+#### What does the escape hatch do?
+
+The escape hatch automatically discards profiling data
+if the ingest limit has been reached.
+
+If the escape hatch activates, it logs the following message:
+
+```text
+Skipping a thread sample period, buffers are full.
+```
+
+You can also look for:
+
+```text
+** THIS WILL RESULT IN LOSS OF PROFILING DATA **.
+```
+
+If you see these log messages, check the exporter implementation.
+
+#### What if I'm on an unsupported .NET version?
+
+None of the .NET Framework versions is supported. You have to switch
+to supported .NET version.
+
+#### Can I tell the sampler to ignore some threads?
+
+There is no such functionality. All managed threads are captured by the profiler.
+
+## Allocation sampling
+
+The profiler samples allocations, captures the call stack state for the .NET
+thread that triggered the allocation, and exporting it in appropriate format.
+
+Use the memory allocation data, together with the stack traces and .NET runtime
+metrics, to investigate memory leaks and unusual consumption patterns
+in Continuous Profiling.
```

Comment:
Would that be Splunk's AlwaysOn Profiling? Since we're upstream, let's change that to:

```suggestion
in an observability back end that supports profiling.
```